src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -7413,18 +7550,12 @@ private static void CombineExtensionMethodArguments(BoundExpression receiver, An
 
                 binder.PopulateExtensionMethodsFromSingleBinder(scope, methodGroup, expression, left, methodName, typeArgumentsWithAnnotations, diagnostics);
 
-                // analyzedArguments will be null if the caller is resolving for error recovery to the first method group
-                // that can accept that receiver, regardless of arguments, when the signature cannot be inferred.
-                // (In the error case of nameof(o.M) or the error case of o.M = null; for instance.)
                 if (analyzedArguments == null)
                 {
-                    if (expression == binder.EnclosingNameofArgument)
```

Comment:
>if (expression == binder.EnclosingNameofArgument) [](http://example.com/codeflow?start=20&length=49)

This deserves a comment because some extension methods might still be applicable to some arguments even though they cannot be reduced based solely on receiver type.

Reply:
I didn't follow. This is for scenarios where we only have a receiver type and no other arguments.

Reply:
There is a change here. It would be good to add a comment explaining why it makes sense to do the filtering unconditionally.

Reply:
I can leave a comment in next PR. We can discuss there